tools/rosparam/src/rosparam/__init__.py

```diff
@@ -631,12 +638,12 @@ def yamlmain(argv=None):
 
 # YAML configuration. Doxygen does not like these being higher up in the code
 
-yaml.add_constructor(u'!radians', construct_angle_radians)
-yaml.add_constructor(u'!degrees', construct_angle_degrees)
+ROSSafeLoader.add_constructor(u'!radians', construct_angle_radians)
+ROSSafeLoader.add_constructor(u'!degrees', construct_angle_degrees)
 
 # allow both !degrees 180, !radians 2*pi
 pattern = re.compile(r'^deg\([^\)]*\)$')
-yaml.add_implicit_resolver(u'!degrees', pattern, first="deg(")
+ROSSafeLoader.add_implicit_resolver(u'!degrees', pattern, first="deg(")
 pattern = re.compile(r'^rad\([^\)]*\)$')
-yaml.add_implicit_resolver(u'!radians', pattern, first="rad(")
+ROSSafeLoader.add_implicit_resolver(u'!radians', pattern, first="rad(")
```

Comment:
Before this change all the added constructors and implicit resolvers affected the global yaml module. If any other existing code relies on that behavior it would break with this change.

Reply:
Good point, I'll adjust it to just modify the global `SafeLoader`. Then I can also remove the added yaml functions that I had added.